nativelib/src/main/scala/scala/scalanative/runtime/Backtrace.scala

```diff
@@ -0,0 +1,236 @@
+        val idx = from + (to - from - 1) / 2
+        val die = dies(idx)
+        if (die.lowPC <= address && address <= die.highPC) Some(die)
+        else if (address < die.lowPC) binarySearch(from, idx)
+        else // die.highPC < address
+          binarySearch(idx + 1, to)
+      }
+    }
+    binarySearch(0, length)
+  }
+
+  private def processMacho(
+      macho: MachO
+  )(implicit bf: BinaryFile): Option[(Vector[DIE], DWARF.Strings)] = {
+    val sections = macho.segments.flatMap(_.sections)
+    for {
+      debug_info <- sections.find(_.sectname == "__debug_info")
+      debug_abbrev <- sections.find(_.sectname == "__debug_abbrev")
+      debug_str <- sections.find(_.sectname == "__debug_str")
+      debug_line <- sections.find(_.sectname == "__debug_line")
+    } yield {
+      readDWARF(
+        debug_info = DWARF.Section(debug_info.offset, debug_info.size),
+        debug_abbrev = DWARF.Section(debug_abbrev.offset, debug_abbrev.size),
+        debug_str = DWARF.Section(debug_str.offset, debug_str.size)
+      )
+    }
+  }
+
+  private def offsetOSX(macho: MachO): Long = {
+    // calculate ASLR offset for macho using vmmap
+    // TODO: retrieve the load address using `mach_vm_region`.
+    // see: https://jvns.ca/blog/2018/01/26/mac-memory-maps/
+    val pid = libc.getpid().intValue()
+    val proc = new ProcessBuilder("vmmap", "-summary", s"$pid").start()
+    val offset = Using(Source.fromInputStream(proc.getInputStream())) { vmmap =>
+      (for {
+        loadAddress <- vmmap
+          .getLines()
+          // "Load Address:    0x0123456789abcdef"
+          .find(_.startsWith("Load Address"))
+          .map { line =>
+            val addressStr =
+              line.split(":").last.trim.drop(2) // drop the prefix 0x
+            java.lang.Long.parseLong(addressStr, 16)
+          }
+        pageZeroSize <- macho.segments
+          .find(_.segname == "__PAGEZERO")
+          .map(_.vmsize)
+      } yield loadAddress - pageZeroSize).getOrElse(0L)
+    }
+    proc.destroy()
+    offset.getOrElse(0L)
+  }
+
+  private def filterSubprograms(dies: Vector[CompileUnit]) = {
+    var filenameAt: Option[UInt] = None
+    dies
+      .flatMap { die =>
+        if (die.is(DWARF.Tag.DW_TAG_subprogram)) {
+          for {
+            line <- die.getLine
+            low <- die.getLowPC
+            high <- die.getHighPC(low)
+          } yield SubprogramDIE(low, high, line, filenameAt)
+        } else if (die.is(DWARF.Tag.DW_TAG_compile_unit)) {
+          // Debug Information Entries (DIE) in DWARF has a tree structure, and
+          // the DIEs after the Compile Unit DIE belongs to that compile unit (file in Scala)
+          // TODO: Parse `.debug_line` section, and decode the filename using
+          // `DW_AT_decl_file` attribute of the `subprogram` DIE.
+          filenameAt = die.getName
+          None
+        } else None
+      }
+      .sortBy(_.lowPC)
+      .toIndexedSeq
+  }
+
+  private def processFile(
```

Comment:
This is the endpoint for parsing the executable, and returns `DwarfInfo` instance.